pkg/provider/instances_v2.go

```diff
@@ -132,20 +132,36 @@ func (i *instancesV2) findInstance(ctx context.Context, fetchers ...InstanceGett
 	return instance, nil
 }
 
-func (i *instancesV2) getNodeAddresses(ifs []kubevirtv1.VirtualMachineInstanceNetworkInterface) []corev1.NodeAddress {
+func (i *instancesV2) getNodeAddresses(ifs []kubevirtv1.VirtualMachineInstanceNetworkInterface, prevAddrs []corev1.NodeAddress) []corev1.NodeAddress {
 	var addrs []corev1.NodeAddress
 
+	foundInternalIP := false
 	// TODO: detect type of all addresses, right now pick only the default
 	for _, i := range ifs {
-		if i.Name == "default" {
+		// Only change the IP if it is known, not if it is empty
+		if i.Name == "default" && i.IP != "" {
```

Comment:
Are we sure this is not affected by link local ipv6 ? so we wrongly enter here when we should not ?

Reply:
We know there's an issue with clearing IPs, but i don't have evidence that link local is causing us issues. I'm not 100% sure we'd be correct in filtering out link local.